application/controllers/admin.php

```diff
@@ -0,0 +1,346 @@
+<?php if ( ! defined('BASEPATH')) exit('No direct script access allowed');
+
+class Admin extends CI_Controller {
+
+	/**
+     * Admin interface for capurace
+     * view, change and export registration info.
+	 *
+	 * 
+	 * 		http://example.com/index.php/a
+	 *	- or -  
+	 * 		http://example.com/index.php/welcome/index
+	 *	- or -
+	 */
+
+    function __construct() {
+        parent::__construct();
+        $this->load->helper('url');
+        $this->load->library('session');
+        $this->tables = array(
+            'group' => array(
+                'group_name' => array(
+                    'type' => 'varchar',
+                    'len' => 10
+                )
+            ),
+            'people' => array(
+                'name' => array(
+                    'type' => 'varchar',
+                    'len' => 10
+                ),
+                'gender' => array(
+                    'type' => 'enum',
+                    'enum' => $GLOBALS['GENDERS']
+                ),
+                'id_card' => array(
+                    'type' => 'varchar',
+                    'len' => 18
+                ),
+                'school_id' => array(
+                    'type' => 'foreignkey',
+                    'references' => 'user',
+                    'display_column' => 'school'
+                ),
+                'accommodation' => array(
+                    'type' => 'boolean'
+                ),
+                'meal' => array(
+                    'type' => 'boolean'
+                ),
+                'race' => array(
+                    'type' => 'enum',
+                    'enum' => $GLOBALS['RACES']
+                )//,
+                //'team_id' => array(
```

Comment:
Ah it was just I didn't know how to join another table multiple times or how to handle cross-referenced foreign key. I'll uncomment them after solving these.